### DISCUSSION TITLE
feat: Allow user to specify categories when using `searxng` for web search

### DISCUSSION THREAD

**Discussion by bmkor** [open-webui/open-webui#8086] [Ideas]
Thanks so much for this wonderful platform. We love this a lot and have been integrating this with our daily work.

Currently, we setup a self-hosted `searxng` engine which was enabled to search our intranet and we did setup a category `intranet` for it. 

It would be great if the websearch can allow user to specify categories for `searxng` so that the search results could all be from the specified categories (such as intranet in our case). In this way, the LLM can more focus on the intranet documents.

To accomplish this, I found out that the [searxng backend in the openweb-ui](https://github.com/open-webui/open-webui/blob/0915352cb075d0a8134904acc87ba755e467aa95/backend/open_webui/retrieval/web/searxng.py#L17) did support the input of categories in the form of kwargs.

I think some tweat on the UI and the setup of ENV variables would easily enable this feature.

If it sounds plausible for you, I am happy to do a PR.

**Comment by oatmealm** [comment]
Sounds very useful. Thanks for sharing this information.

**Reply by bmkor** [reply]
Think it would be useful when creating a custom model. By the way, I would like to try a PR. May I?
